projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts

```diff
@@ -0,0 +1,32 @@
+import { OnInit, ElementRef } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Product, ProductService, RoutingService } from '@spartacus/core';
+import { filter, map, switchMap } from 'rxjs/operators';
+
+export abstract class BaseyotpoComponent implements OnInit {
+  product$: Observable<Product>;
+
+  constructor(protected elementRef:ElementRef, 
+		protected routingService: RoutingService,
+		protected productService: ProductService) { }
+
+  ngOnInit() {
+    this.product$ = this.getProduct();
+  }
+
+  getProduct(): Observable<Product> {
```

Comment:
Typically component inheritance is not encouraged in angular. You should move the shared logic to a component that is injected in both Yotpo components. Like the `CurrentProductService`.

Reply:
fixed